h2/config.py

```diff
@@ -25,6 +26,21 @@ def __set__(self, instance, value):
         setattr(instance, self.attr_name, value)
 
 
+class DummyLogger(logging.Logger):
+    """
+    An Logger object that does not actual logging, hence a DummyLogger.
+
+    For the class the log operation is merely a no-op.  The intent is to avoid
+    conditionals being sprinkled throughout the hyper-h2 code for calls to
+    logging functions when no logger is passed into the corresponding object.
+    """
+    def _log(self, *vargs, **kwargs):
+        """
+        No-op logging.
+        """
+        pass
```

Comment:
So, the short answer is that this relies on internal implementation details of the logging module: specifically, we are relying on the idea that the logger will always call `_log`.

That's *likely*, but not guaranteed, and if that guarantee went away the failure mode would be very subtle. It is much safer to patch the interface we actually use: that way, we know that in the most-common case (no logging attached) we aren't going to get unexpectedly tripped up.